beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -3645,12 +3645,13 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             ProduceBlockVerification::VerifyRandao => BlockSignatureStrategy::VerifyRandao,
             ProduceBlockVerification::NoVerification => BlockSignatureStrategy::NoVerification,
         };
+        let mut ctxt = ConsensusContext::new(block.slot()).set_proposer_index(proposer_index);
```

Comment:
On second thoughts we probably shouldn't set the proposer index during block production, as it is desired (see: https://github.com/sigp/lighthouse/pull/2740)

The context will take care of calculating it only once